bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/ConfigDetails.java

```diff
@@ -16,22 +16,25 @@
  * under the License.
  */
 
-package io.ballerina.runtime.internal.configurable.providers.toml;
+package io.ballerina.runtime.internal.configurable.providers;
 
 import java.nio.file.Path;
+import java.util.Map;
 
 /**
  * Represents information on Configurable TOML data.
  *
  * @since 2.0
  */
-public class TomlDetails {
+public class ConfigDetails {
 
     public final Path[] paths;
     public final String configContent;
+    public final Map<String, String> envVariables;
 
-    public TomlDetails(Path[] paths, String configContent) {
+    public ConfigDetails(Path[] paths, String configContent, Map<String, String> envVariables) {
         this.paths = paths;
         this.configContent = configContent;
+        this.envVariables = envVariables;
```

Comment:
We need to think again do we need to provide the env variables from outside.

Reply:
Addressed in commit https://github.com/ballerina-platform/ballerina-lang/pull/42115/commits/3ad49d243d8c5e18f0859d0fecbf74a475029e9b